src/components/UploadingJobsDisplay/UploadingJobsDisplay.js

```diff
@@ -48,6 +48,29 @@ class UploadingJobsDisplay extends Component {
     this.uploadJobs();
   }
 
+  componentDidUpdate() {
+    if (!this.props.files) {
+      return;
+    }
+
+    let dontLeave = false;
+
+    for (const file in this.props.files) {
+      if (this.props.files[file].status.includes('Uploading')) {
+        dontLeave = true;
+        break;
+      }
+    }
+    // prevent from leaving the page in case of download in progress
+    if (dontLeave) {
+      window.onbeforeunload = () => {
```

Comment:
This code does not cover the case when there is the file in process of uploading and the user navigates away to another page within the folio application.

Reply:
It should be OK as download process will not be interrupted by a browser.